### DISCUSSION TITLE
Use custom Auth context instead of the one supplied by Supabase UI.

### DISCUSSION THREAD

**Discussion by YannHulot** [Etheonor/supanextail#22] [Feedback]
So to detect if a user is logged in or not, you use:

` const { user, session } = Auth.useUser();`

`Auth` is this case comes from '@supabase/ui' which according to the docs is not yet ready for production. The second argument for not using this hook is that you are importing the `Auth.useUser` hook from a UI library. 

You are already using `daisy-ui` so it does not make sense to have two competing UI libraries imported in the same project. 
Also it looks like that `@supabase/ui` is not being tree-shaken properly as the weight of this dependency is 20k after GZIP. 

In this case, it makes more sense to use a custom context that detect if the user is logged in and will return the user data once a user logs in. 

The session will still be saved on the users browser and you will be able to enforce row level security nonetheless.

This article is a good starting point on how to create and integrate a custom auth context within a React app: https://dev.to/finiam/predictable-react-authentication-with-the-context-api-g10

Finally having a custom auth context gives you more flexibility regarding the actions that a user can do and their side effects.

**Comment by himeshp** [comment]
with custom auth, achieving row-level security in supabase will be a challenge if I am not wrong.

**Comment by Etheonor** [comment]
It will be my next task. I'm not a fan of the supabase UI component anyway, and I already did that in another project.

**Reply by himeshp** [reply]
supabase is launching updated auth v2 this week, might be worth a wait before u decide to hop for custom auth code.

**Reply by YannHulot** [reply]
agreed, wait and see if it fits your needs @Etheonor . All the discussions I have created are merely suggestions. I dont pretend that I know what I am doing.

**Reply by Etheonor** [reply]
ahah no problem, I really like your suggestions ^^ Didn't see the Supabase week so I'll wait for their announcements

**Comment by Etheonor** [comment]
Well Auth 2.0 is basically phone auth. Even though I think it's really interesting, it's not the improvements we were looking for. 

So custom auth it is! I'll prioritize Typescript first as I don't want to refactor twice.

**Reply by Etheonor** [reply]
Done
